lib/balance-tx/lib/internal/Internal/Cardano/Write/Tx/Balance.hs

```diff
@@ -916,6 +884,13 @@ balanceTransactionInner
         left ErrBalanceTxAssignRedeemers $
             assignScriptRedeemers pp timeTranslation utxoReference redeemers tx'
 
+data Selection = Selection
+    { extraInputs :: [W.TxIn]
+    , extraCollateral :: [W.TxIn]
+    , extraOutputs :: [W.TxOut]
+    , extraInputScripts :: [CA.Script CA.KeyHash]
+    } deriving (Eq, Show)
```

Comment:
> Is it correct that the only difference between this new Selection type and the existing TxUpdate type is the absence/presence of the feeUpdate field?

Yes.

> strong reason for not just returning a TxUpdate from selectAssets, and then using updateTx to apply the result?

I was previously considering making `TxUpdate` contain `Selection`, but this might be an option as well. I will think about it.

Reply:
In that case, can I propose that for this PR:

- we rename the `Selection` data type to `TxUpdateWithoutFee` (or similar).
- we add a comment with a TODO reminding us to reconcile the `TxUpdate` and `TxUpdateWithoutFee` types, with a link to a ticket?

My main issue with the name `Selection` is that this term is already rather overloaded. In particular, it has a well-defined meaning in the context of coin selection: a prototype for a complete transaction, rather than an update.

Reply:
Maybe let's discuss on tuesday. I was rather fond of naming it `Selection`, as the function retruning it it is exactly `selectAssets`, which is the (top-most) interface between `balanceTx` and `coin-selection`. Shadowing e.g. `Cardano.CoinSelection.Selection` is, I guess to some extent, the very point.

Reply:
Alternative suggestion: use **_ledger types_** in our definition of `Selection`, so that we can return everything. This way, the `balance-tx` notion of a `Selection` will be similar to the `coin-selection` library's notion of a `Selection`.

```hs
data Selection era = Selection
    { inputs :: [TxIn]
    -- ^ all inputs, including those pre-selected by the caller
    , collateralInputs :: [TxIn]
    -- ^ all collateral inputs, including those pre-selected by the caller
    , outputs :: [TxOut era]
    -- ^ all outputs, including change outputs and those pre-defined by the caller
    , inputScripts :: [CA.Script CA.KeyHash]
    }
```